benchopt/cli/tests/test_cli.py

```diff
@@ -174,6 +177,20 @@ def test_timeout_in_env(self, test_env_name, timeout):
         # Make sure the results were saved in a result file
         assert len(out.result_files) == 1, out.output
 
+    def test_no_timeout(self):
+        old_value = str(get_setting('default_timeout'))
+        os.environ['BENCHOPT_DEFAULT_TIMEOUT'] = "0"
+        with CaptureRunOutput() as out:
+            try:
+                run([str(DUMMY_BENCHMARK_PATH), '-d', SELECT_ONE_SIMULATED, '-f', 
+                    SELECT_ONE_PGD, '-o',SELECT_ONE_OBJECTIVE, '--no-plot', '--no-timeout'], 
+                    'benchopt', standalone_mode=False)
+            finally:
+                os.environ['BENCHOPT_DEFAULT_TIMEOUT'] = old_value
+
+        out.check_output('timeout', repetition=0)
```

Comment:
To make sure this is working properly, it would be good to have first a run that timeouts (without  `--no-timeout`) and then check that adding the `--no-timeout` works properly.

Reply:
Done!